### PR TITLE
source-squarespace contribution from avirajsingh7

### LINKED ISSUE #46068: [connector-request] Squarespace
This is a **suggestion** to create a connector for **Squarespace** using the UI Connector Builder.
The website for the connector is https://www.squarespace.com.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by avirajsingh7**
Closing issue : #46068 

## What

This PR adds a new connector for source Squarespace (source-squarespace).

Contributor-provided connector description:

The Squarespace connector enables seamless integration with your Squarespace store, allowing you to sync data from various key endpoints
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `api_key` | `string` | API Key. API key to use. Find it at https://developers.squarespace.com/commerce-apis/authentication-and-permissions |  |
| `start_date` | `string` | Start date. Any data before this date will not be replicated. |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| inventory | variantId | DefaultPaginator | ✅ |  ❌  |
| store_pages | id | DefaultPaginator | ✅ |  ❌  |
| products | id | DefaultPaginator | ✅ |  ✅  |
| profiles | id | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: airbyte-integrations/connectors/source-squarespace/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,1664 @@
...
+            type: RequestPath
+          pagination_strategy:
+            type: CursorPagination
+            cursor_value: "{{ response.get(\"pagination\", {}).get(\"nextPageUrl\", {}) }}"
+            stop_condition: "{{ not response.get(\"pagination\", {}).get(\"nextPageUrl\", {}) }}"
+      schema_loader:
+        type: InlineSchemaLoader
+        schema:
+          $ref: "#/schemas/inventory"
+    orders:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-squarespace/manifest.yaml:45]: This isn't described in the docs please check that.

File: docs/integrations/sources/squarespace.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+| `api_key` | `string` | API Key. API key to use. Find it at https://developers.squarespace.com/commerce-apis/authentication-and-permissions |  |
+| `start_date` | `string` | Start date. Any data before this date will not be replicated. |  |
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| inventory | variantId | DefaultPaginator | ✅ |  ❌  |
+| store_pages | id | DefaultPaginator | ✅ |  ❌  |
+| products | id | DefaultPaginator | ✅ |  ✅  |
+| profiles | id | DefaultPaginator | ✅ |  ❌  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/squarespace.md:17]: Missing some streams.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
@avirajsingh7 did you have time to take a look in my comments?

File: docs/integrations/sources/squarespace.md (hunk 2 above)
Comment (avirajsingh7) [reply docs/integrations/sources/squarespace.md:17]: @marcosmarxm  done added

**Comment by avirajsingh7** [comment]
@marcosmarxm sorry for delay, I have updated a docs

**Review by marcosmarxm — APPROVED** [review]
Great contribution @avirajsingh7

**Pull request merged**